cgltf.h

```diff
@@ -128,6 +119,18 @@ typedef enum cgltf_result
 	cgltf_result_legacy_gltf,
 } cgltf_result;
 
+typedef struct cgltf_options
+{
+	cgltf_file_type type; /* invalid == auto detect */
+	cgltf_size json_token_count; /* 0 == auto */
+	void* (*memory_alloc)(void* user, cgltf_size size);
+	void (*memory_free) (void* user, void* ptr);
+	void* memory_user_data;
+	cgltf_result (*file_read)(const struct cgltf_options* options, const char* path, cgltf_size* size, void** data);
+	void (*file_release)(void (*memory_free) (void* user, void* ptr), void* memory_user_data, void* file_user_data, void* data);
```

Comment:
Passing `options` to `file_read()` but the relevant members thereof to `file_release()` looks a little bit asymmetric.

Reply:
You're completely right, I was bothered by that as well. However, we don't have `cgltf_options` around anymore when `cgltf_free()` is called by the user. And I think all other options (e.g. putting the callback and user data into a new struct and having that in `cgltf_options`, etc.) would produce more complexity. Or do you have a good idea?

Reply:
The default implementation of `file_read()` needs `cgltf_options` in order to be able to respect overridden memory allocation. But I don't feel it would be necessary to provide `cgltf_options` to a custom `file_*` implementation, it should be able to manage that through `file_user_data`. A custom `file_*` implementation shouldn't need the `memory_*` parameters in `file_release()`. So what complicates matters is the default implementation of `file_*` which doesn't have anywhere to tuck away custom memory allocation parameters, like an override can.

One option could be for the default file read implementation not to be statically available, that you would need to allocate it's `file_user_data` (and perhaps drop "user" in the name) given a set of `cgltf_options`. And then of course free it at the end. In other words you would have to instantiate a default implementation if you want to be able to read files on the fly. Not wildly elegant?

Maybe individual structs for each type of override (`cgltf_memory_options` and `cgltf_file_options`) inside `cgltf_options` isn't such a bad idea after all? Then you could pass both to `file_read()` and only `cgltf_memory_options` to `file_release()`. If you see more overrides and callbacks along these lines coming in the future (perhaps for extension parsing/hooks?) then isolating the overrides might be a good idea.

I might have made a hen out of a feather here.. :-P

Reply:
You kind of followed the dilemma I had while implementing this very well here. :-) I think in the end I'd prefer the `cgltf_memory_options` and `cgltf_file_options` approach...
What do you think about the updated implementation?

Reply:
I would say it's looking really good now!